Index: dawn-2025.11.07~7414c34a/src/network/multicastsocket.c
===================================================================
--- dawn-2025.11.07~7414c34a.orig/src/network/multicastsocket.c
+++ dawn-2025.11.07~7414c34a/src/network/multicastsocket.c
@@ -68,9 +68,14 @@ int setup_multicast_socket(const char *_
                    IPPROTO_IP,
                    IP_ADD_MEMBERSHIP,
                    &command, sizeof(command)) < 0) {
-        dawnlog_perror("setsockopt:IP_ADD_MEMBERSHIP");
+    	if(errno == ENODEV)
+    		dawnlog_warning("No interface available for multicast");
+    	else
+    		dawnlog_perror("setsockopt:IP_ADD_MEMBERSHIP");
+
         // FIXME: Is close() required?
         close(sock);
+        exit(EXIT_FAILURE);
     }
     return sock;
 }
Index: dawn-2025.11.07~7414c34a/src/storage/datastorage.c
===================================================================
--- dawn-2025.11.07~7414c34a.orig/src/storage/datastorage.c
+++ dawn-2025.11.07~7414c34a/src/storage/datastorage.c
@@ -180,8 +180,8 @@ void send_beacon_requests(ap* target_ap,
         if (i->rrm_enabled_capa & dawn_metric.rrm_mode_mask)
         {
             int ubus_status = ubus_send_beacon_request(i, target_ap, dawn_metric.duration, sub_id);
-            if (ubus_status) /* change to info since this can happen if devices go offline and we dont want to flood logs */
-                dawnlog_info("Client / BSSID = " MACSTR " / " MACSTR ": BEACON REQUEST failed",
+            if (ubus_status)
+                dawnlog_warning("Client / BSSID = " MACSTR " / " MACSTR ": BEACON REQUEST failed",
                     MAC2STR(i->client_addr.u8), MAC2STR(target_ap->bssid_addr.u8));
         }
 
Index: dawn-2025.11.07~7414c34a/src/network/networksocket.c
===================================================================
--- dawn-2025.11.07~7414c34a.orig/src/network/networksocket.c
+++ dawn-2025.11.07~7414c34a/src/network/networksocket.c
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <errno.h>
 #include <libubox/blobmsg_json.h>
 
 #include "utils.h"
@@ -70,7 +71,7 @@ static void* receive_msg_inner(void* arg
 
         //FIXME: Next few lines look a bit mangled, with odd strlen() tests, etc...
         if (recv_string_len < 0) {
-            dawnlog_error("Could not receive message!");
+            dawnlog_error("Could not receive message!: %d: %s", errno, strerror(errno));
             continue;
         }
 
Index: dawn-2025.11.07~7414c34a/src/utils/ubus.c
===================================================================
--- dawn-2025.11.07~7414c34a.orig/src/utils/ubus.c
+++ dawn-2025.11.07~7414c34a/src/utils/ubus.c
@@ -1697,17 +1697,11 @@ bool subscribe(struct ubus_context *ctx_
 
     sprintf(subscribe_name, "hostapd.%s", hostapd_entry->iface_name);
 
-    if (ubus_lookup_id(ctx_local, subscribe_name, &hostapd_entry->id)) {
-        dawnlog_warning("Failed to lookup ID!");
-        subscription_wait(&hostapd_entry->wait_handler);
-        return false;
-    }
-
-    if (ubus_subscribe(ctx_local, &hostapd_entry->subscriber, hostapd_entry->id)) {
-        dawnlog_warning("Failed to register subscriber!");
-        subscription_wait(&hostapd_entry->wait_handler);
-        return false;
-    }
+    if (ubus_lookup_id(ctx_local, subscribe_name, &hostapd_entry->id))
+    	goto subscription_wait;
+
+    if (ubus_subscribe(ctx_local, &hostapd_entry->subscriber, hostapd_entry->id))
+    	goto subscription_wait;
 
     hostapd_entry->subscribed = true;
 
@@ -1726,6 +1720,10 @@ bool subscribe(struct ubus_context *ctx_
     dawnlog_debug("Subscribed to: %s\n", hostapd_entry->iface_name);
 
     return true;
+subscription_wait:
+	dawnlog_warning("radio interface %s service not ready yet", hostapd_entry->iface_name);
+	subscription_wait(&hostapd_entry->wait_handler);
+	return false;
 }
 
 static void
@@ -1758,7 +1756,8 @@ wait_cb(struct ubus_context *ctx_local,
     if (strcmp(sub->iface_name, path + 1))
         return;
 
-    subscribe(ctx_local, sub);
+    if(subscribe(ctx_local, sub))
+    	dawnlog_warning("radio interface %s service subscribed", sub->iface_name);
 }
 
 bool subscriber_to_interface(const char *ifname) {
